test/functional/fixtures/runner/upload/upload-test.js

```diff
@@ -0,0 +1,97 @@
+var expect    = require('chai').expect;
+var fs        = require('fs');
+var promisify = require('../../../../../lib/utils/promisify.js');
+
+
+var readFile     = promisify(fs.readFile);
+var srcFilePath1 = 'test/functional/fixtures/runner/upload/files/text1.txt';
+var srcFilePath2 = 'test/functional/fixtures/runner/upload/files/text2.txt';
+
+
+function compareWithUploadedFiles (originalFile) {
+    if (!uploadedFiles)
+        throw new Error('There are no files uploaded');
+
+    expect(browsersInfo.length).eql(uploadedFiles.length,
+        'The number of uploaded files is different from the number of the browsers');
+
+    uploadedFiles.forEach(function (file) {
+        expect(file).eql(originalFile);
+    });
+
+    delete global.uploadedFiles;
+}
+
+
+describe('Upload', function () {
+    it('Should success when uploading file', function () {
+        return runTests('upload.test.js', 'Upload a file')
+            .then(function (err) {
+                expect(err).eql('');
+
+                return readFile(srcFilePath1)
+                    .then(function (originalFile) {
+                        return compareWithUploadedFiles(originalFile);
+                    });
+            });
+    });
+
+    it('Should success when uploading multiple files', function () {
+        return runTests('upload.test.js', 'Upload multiple files')
+            .then(function (err) {
+                expect(err).eql('');
+
+                return Promise
+                    .all([readFile(srcFilePath1), readFile(srcFilePath2)])
+                    .then(function (originalFiles) {
+                        return compareWithUploadedFiles(originalFiles);
+                    });
+            });
+    });
+
+    it('Should fail when uploading a non-existent file', function () {
+        return runTests('upload.test.js', 'Upload a non-existent file - should fail', { shouldFail: true })
+            .catch(function (err) {
+                expect(err).contains('fake.jpg');
+            });
+    });
+
+    it('Should fail when uploading multiple files including non-existent files', function () {
+        return runTests('upload.test.js', 'Upload multiple files - should fail', { shouldFail: true })
+            .catch(function (err) {
+                expect(err).contains('fake1.jpg');
+                expect(err).contains('fake2.jpg');
+                expect(err).not.contains('text1.txt');
+            });
+    });
+
+    describe('regression', function () {
+        it('Should pass when attaching files with an added input', function () {
+            return runTests('regression.test.js', 'Upload for an added element')
+                .then(function (err) {
+                    expect(err).eql('');
+                });
+        });
+
+        it('Should pass when attaching files with a replaced input', function () {
+            return runTests('regression.test.js', 'Upload for a replaced element')

```

Comment:
Replace on the following scenario:
1)there are two inputs: `input1` and `input2` inside a one form.
2)for `input1` is selected file1.txt, for `input2` - file2.txt
3)using javascript perform the replacing:
`input1`.parentNode.replaceChild(`input1`, `input2`).
4)upload the form
5)check that was uploaded only file1.txt (from `input1`)

For more details see https://github.com/DevExpress/testcafe-hammerhead/issues/399